prime-router/src/test/kotlin/ReportTests.kt

```diff
@@ -909,128 +933,288 @@ class ReportTests {
+            }/devstoreaccount1;QueueEndpoint=http://${azuriteContainer.host}:${
+                azuriteContainer.getMappedPort(
+                    10001
+                )
+            }/devstoreaccount1;"""
+        val blobContainerMetadata = BlobAccess.BlobContainerMetadata(
+            "container1",
+            blobConnectionString
+        )
+
+        mockkObject(BlobAccess.BlobContainerMetadata.Companion)
+        every { BlobAccess.BlobContainerMetadata.Companion.build(any(), any()) } returns blobContainerMetadata
+        every { BlobAccess.BlobContainerMetadata.Companion.build(any()) } returns blobContainerMetadata
+
         val mockMetadata = mockk<Metadata> {
             every { fileNameTemplates } returns emptyMap()
         }
         val mockActionHistory = mockk<ActionHistory> {
             every { trackCreatedReport(any(), any(), blobInfo = any()) } returns Unit
         }
-        val hl7MockData = UUID.randomUUID().toString().toByteArray() // Just some data
+        val fhirMockData = UUID.randomUUID().toString().toByteArray() // Just some data
         val receiver = Receiver(
             "name", "org", Topic.FULL_ELR,
-            translation = Hl7Configuration(
-                receivingApplicationName = null, receivingApplicationOID = null,
-                receivingFacilityName = null, receivingFacilityOID = null, receivingOrganization = null,
-                messageProfileId = null
-            )
+            translation = FHIRConfiguration(receivingOrganization = null)
         )
 
         // Now test single report
-        mockkObject(BlobAccess)
-        every {
-            BlobAccess.uploadBody(
-                Report.Format.HL7, hl7MockData, any(), any(), Event.EventAction.PROCESS
-            )
-        } returns
-            BlobAccess.BlobInfo(Report.Format.HL7, "someurl", "digest".toByteArray())
-
         var reportIds = listOf(ReportId.randomUUID())
         val (report, event, blobInfo) = Report.generateReportAndUploadBlob(
-            Event.EventAction.PROCESS, hl7MockData, reportIds, receiver, mockMetadata, mockActionHistory,
+            Event.EventAction.PROCESS, fhirMockData, reportIds, receiver, mockMetadata, mockActionHistory,
             topic = Topic.FULL_ELR,
         )
-        unmockkObject(BlobAccess)
 
-        assertThat(report.bodyFormat).isEqualTo(Report.Format.HL7)
+        assertThat(report.bodyFormat).isEqualTo(Report.Format.FHIR)
         assertThat(report.itemCount).isEqualTo(1)
         assertThat(report.destination).isNotNull()
         assertThat(report.destination!!.name).isEqualTo(receiver.name)
         assertThat(report.itemLineages).isNotNull()
         assertThat(report.itemLineages!!.size).isEqualTo(1)
         assertThat(event.eventAction).isEqualTo(Event.EventAction.PROCESS)
-        assertThat(blobInfo.blobUrl).isEqualTo("someurl")
+        assertThat(blobInfo.blobUrl).endsWith("/devstoreaccount1/container1/process%2Forg.name%2F${report.name}")
```

Comment:
You could further enhance this by downloading the blob too and checking the contents equals